File: src/integrationMongoTest/java/io/lsdconsulting/lsd/distributed/interceptor/integration/listener/MongoDbTestListener.java
package io.lsdconsulting.lsd.distributed.interceptor.integration.listener;

import io.lsdconsulting.lsd.distributed.interceptor.integration.testapp.repository.TestRepository;
import lombok.extern.slf4j.Slf4j;
import org.junit.platform.launcher.TestExecutionListener;
import org.junit.platform.launcher.TestPlan;

@Slf4j
public class MongoDbTestListener implements TestExecutionListener {

    public void testPlanExecutionStarted(TestPlan testPlan) {
        log.info("MongoDbTestListener.testPlanExecutionStarted");
        TestRepository.setupDatabase();
    }

    public void testPlanExecutionFinished(TestPlan testPlan) {
        log.info("MongoDbTestListener.testPlanExecutionFinished");
        TestRepository.tearDownDatabase();
    }
}

File: integrationMongoTest/src/test/kotlin/io/lsdconsulting/lsd/distributed/interceptor/integration/listener/MongoDbTestListener.kt
package io.lsdconsulting.lsd.distributed.interceptor.integration.listener

import io.lsdconsulting.lsd.distributed.interceptor.integration.testapp.repository.TestRepository
import lsd.logging.log
import org.junit.platform.launcher.TestExecutionListener
import org.junit.platform.launcher.TestPlan

class MongoDbTestListener : TestExecutionListener {
    override fun testPlanExecutionStarted(testPlan: TestPlan) {
        log().info("MongoDbTestListener.testPlanExecutionStarted")
        TestRepository.setupDatabase()
    }

    override fun testPlanExecutionFinished(testPlan: TestPlan) {
        log().info("MongoDbTestListener.testPlanExecutionFinished")
        TestRepository.tearDownDatabase()
    }
}
